docs/csharp/language-reference/operators/boolean-logical-operators.md

```diff
@@ -0,0 +1,159 @@
+---
+title: "Boolean logical operators - C# Reference"
+description: "Learn about C# operators that perform logical negation, conjunction (AND), and inclusive and exclusive disjunction (OR) operations with Boolean operands."
+ms.date: 04/05/2019
+author: pkulikov
+f1_keywords: 
+  - "!_CSharpKeyword"
+  - "&_CSharpKeyword"
+  - "^_CSharpKeyword"
+  - "|_CSharpKeyword"
+  - "&&_CSharpKeyword"
+  - "||_CSharpKeyword"
+helpviewer_keywords: 
+  - "logical operators [C#]"
+  - "short-circuiting operators [C#]"
+  - "logical negation operator [C#]"
+  - "NOT operator [C#]"
+  - "! operator [C#]"
+  - "AND operator [C#]"
+  - "ampersand operator [C#]"
+  - "conjunction operator [C#]"
+  - "& operator [C#]"
+  - "exclusive OR operator [C#]"
+  - "XOR operator [C#]"
+  - "^ operator [C#]"
+  - "OR operator [C#]"
+  - "disjunction operator [C#]"
+  - "| operator [C#]"
+  - "conditional logical AND operator [C#]"
+  - "short-circuiting logical AND operator [C#]"
+  - "&& operator [C#]"
+  - "conditional logical OR operator [C#]"
+  - "short-circuiting logical OR operator [C#]"
+  - "|| operator [C#]"
+---
+# Boolean logical operators (C# Reference)
+
+The following operators perform logical operations with [bool](../keywords/bool.md) operands:
+
+- Unary [`!` (logical negation)](#logical-negation-operator-) operator.
+- Binary [`&` (logical AND)](#logical-and-operator-), [`|` (logical OR)](#logical-or-operator-), and [`^` (logical exclusive OR)](#logical-exclusive-or-operator-) operators. Those operators always evaluate both operands.
+- Binary [`&&` (conditional logical AND)](#conditional-logical-and-operator-) and [`||` (conditional logical OR)](#conditional-logical-or-operator-) operators. Those operators evaluate the second operand only if it's necessary.
+
+The `&` and `|` operators are also defined for `bool?` operands. For more information, see [The bool? type](../../programming-guide/nullable-types/using-nullable-types.md#the-bool-type) section of the [Using nullable types](../../programming-guide/nullable-types/using-nullable-types.md) article.
+
+For the operands of [integral](../keywords/integral-types-table.md) types, the `&`, `|`, and `^` operators perform bitwise logical operations.
+
+## Logical negation operator !
+
+The unary `!` operator computes logical negation of its operand. That is, it produces `true`, if the operand evaluates to `false`, and `false`, if the operand evaluates to `true`:
+
+[!code-csharp-interactive[logical negation](~/samples/snippets/csharp/language-reference/operators/LogicalOperators.cs#Negation)]
+
+## Logical AND operator &amp;
```

Comment:
@BillWagner could you please tell me what anchor is generated for this header on the review site. I use `logical-and-operator-` in this article (does it work?); however, Visual Studio Code doesn't recognize such a link and doesn't navigate me to the section.

Reply:
The anchor in the review site is `logical-and-operator-` .  Good guess!

Reply:
Cool, thanks!